.github/workflows/artifact-tests.yml

```diff
@@ -0,0 +1,79 @@
+name: artifact-unit-tests
+on: push
+
+jobs:
+  build:
+    name: Build
+
+    strategy:
+      matrix:
+        runs-on: [ubuntu-latest, windows-latest, macOS-latest]
+      fail-fast: false
+
+    runs-on: ${{ matrix.runs-on }}
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Set Node.js 12.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: 12.x
+
+    # In order to upload & download artifacts from a shell script, certain env variables need to be set that are only available in the
+    # node context. This runs a local action that gets and sets the necessary env variables that are needed
+    - name: Set env variables
+      uses: ./packages/artifact/__tests__/ci-test-action/
+
+    # Need root node_modules because certain npm packages like jest are configured for the entire repository and it won't be possible
+    # without these to just compile the artifacts package
+    - name: Install root npm packages
+      run: npm ci
+
+    - name: Compile artifact package
+      run: |
+        npm ci
+        npm run tsc
+      working-directory: packages/artifact
+    
+    - name: Set artifact file contents
+      run: |
+        echo "::set-env name=non-gzip-artifact-content::hello"
+        echo "::set-env name=gzip-artifact-content::Some large amount of text that has a compression ratio that is greater than 100%. If greater than 100%, gzip is used to upload the file"
+
+    - name: Create files that will be uploaded
+      run: |
+        mkdir artifact-path 
+        echo ${{ env.non-gzip-artifact-content }} > artifact-path/world.txt
+        echo ${{ env.gzip-artifact-content }} > artifact-path/gzip.txt
```

Comment:
I recommend:

1. Using a file/file comparison instead of a file/string comparison for your verification. You could pass these `artifact-path` file paths to `test-artifact-file.sh`, or possibly `cmp` directly, below.

2. Skipping the use of environment variables here - just use the following:

```
        echo "hello" > artifact-path/world.txt
        echo "Some large amount of text that has a compression ratio that is greater than 100%. If greater than 100%, gzip is used to upload the file" > artifact-path/gzip.txt
```

Reply:
The file comparison is a good idea, same with the multi-file artifact. I'll work on that in a subsequent PR. 

Env variables were used to cut down on copy and pasting since later on they're used to check the contents of the downloaded files

Reply:
If we switch to file comparisons, then we'll no longer need the environment variables. 🎉